lib/coinbase/staking_balance.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'date'
+
+module Coinbase
+  # A representation of a staking balance on a network for a given asset.
+  class StakingBalance
+    # Returns a list of StakingBalance for the provided network, asset, and addresses.
+    # @param network_id [Symbol] The network ID
+    # @param asset_id [Symbol] The asset ID
+    # @param address_ids [Array<String>] The address IDs
```

Comment:
unlike staking rewards, staking balance query accepts only 1 address at a time, our rewards client will report this error when querying with `address=a or address=b`
```
{
  "code": "bad_staking_request",
  "message": "Filter validation failed. invalid filter provided: filter only accepts additive expressions (i.e. using AND). using OR, NOT, or other expressions isn't supported. please use only AND expressions"
}
```